### PR TITLE
Fixes for phoneutria, solidarity badge, seymour/audrey, federal fundraising

### LINKED ISSUE #6966: Epiphany shows No Thanks on top of R&D
Link to bug replay: https://jinteki.net/bug-report/e97e688c-2715-49e8-913d-afb2caa8fe31?b=0

image

### LINKED ISSUE #6967: Shipment from Vladisibirsk shows wrong message
Link to bug replay: https://jinteki.net/bug-report/e97e688c-2715-49e8-913d-afb2caa8fe31?b=0

Shipment from Vladisibirsk claims to trash all cards from HQ.

### LINKED ISSUE #6970: Solidarity Badge triggers on second trash
Link to bug replay: https://jinteki.net/bug-report/014f2cef-06cf-4321-9204-c989b44a652d?b=0

Description:

Solidarity Badge triggers on second trash in a turn.

### LINKED ISSUE #6971: Phoneutria ability triggering on all ice
If Phoneutria is rezzed on any server, the tag ability gets triggered every time you pass any ice on the board, even unrezzed.

### LINKED ISSUE #6974: Federal Fundraising seems to trigger after mandatory draw
I'm not 100% sure on this one because I had a whole bunch of start of turn triggers firing at once, but it seems to me as if Federal Fundraising triggers its draw effect after mandatory draw. It might have behaved differently in cases where I declined to look at the top 3 cards; I think it might have triggered correctly before mandatory draw if I did look at the top, but after if I didn't, but my memory is not perfect. I also had 2x DBS running through all of this so I was looking at a lot of cards and that might have confused things somewhat, but I know for sure that I on multiple occasions drew my mandatory and then got the "draw a card?" prompt from fundraising.

### LINKED ISSUE #6976: Skimming doesn't show the card in the log
I'm working on it

### LINKED ISSUE #6977: Tatu bola swapping doesn't trigger install events
I'm working on it. This one kind of changed based on the nsg ruling.

### LINKED ISSUE #6979: Tucana on steal effect didn't trigger when it was trashed first
Tucana was in a server. The Runner trashed Tucana first, and then stole an agenda in the same server. Tucana did not trigger, despite its effect being a Persistent effect.

### LINKED ISSUE #6980: Space needed in chat log for Epiphany
XXX uses Epiphany Analytica: Nations Undivided to install the thirdcard from R&D.

space needed between the index (first, second, third) and card.

### LINKED ISSUE #6981: Space needed in Federal Fundraising chat log
XXXX uses Federal Fundraising to place a card ontop of R&D.

"on top"

### LINKED ISSUE #6984: Eru Ayase-Pessoa incorrectly triggering threat condition
Was playing a game as Ob against an Anarch player and Esu triggered the runner accessing two cards despite us each only having 2 agenda points. No screenshots but game log attached.

Casual format game
Browser: Edge
OS: Win11
[gamelog.txt](https://github.com/mtgred/netrunner/files/12247406/gamelog.txt)

### PR DESCRIPTION
**Pull request by NBKelly**
Closes #6971
Closes #6970
Closes #6974 (it had no async on the decline to look, so it lagged behind the draw)
Closes #6967 
fixed text on Audrey v2

Closes #6984 
Closes #6977 
Closes #6976 
Closes #6966 
Closes #6979 
Closes #6980 
Closes #6981

### CODE REVIEW

**Comment by NBKelly** [comment]
Also closes #6967

File: src/clj/game/core/moving.clj [hunk 1]
```diff
@@ -497,8 +497,16 @@
...
+                         :swap true})
+          ;; under the new nsg rules, swapping a card into play triggers an install event
+          ;; TODO - quote exactly where
+          install-event (or (and (installed? a) (not (installed? b)))
+                            (and (installed? b) (not (installed? a))))]
       (trigger-event state side :swap moved-a moved-b)
+      (when (and install-event (= :corp side))
+        (trigger-event state side :corp-install
+                       {:card (get-card state (if (installed? moved-a) moved-a moved-b))
+                        :install-state (:install-state (card-def (if (installed? moved-a) moved-a moved-b)))}))
```
Comment (NoahTheDuke) [inline src/clj/game/core/moving.clj:509]: This doesn't work and can't work here. `:corp-install` must be async (takes an eid), and this function isn't. Either do the extremely annoying and arduous task of changing all uses of `swap-cards` to be async (might not be possible), or create a new async (takes an eid) function `swap-into-play` that you use in the right card impl.

**Review by NoahTheDuke — CHANGES_REQUESTED** [review]

File: src/clj/game/core/moving.clj [hunk 2]
```diff
@@ -518,6 +514,24 @@
         (when (in-hand? moved-b) (add-to-currently-drawing state b-side moved-b)))
       [(get-card state moved-a) (get-card state moved-b)])))
 
+(defn swap-cards-async
+  "Swaps two cards when one or both aren't installed"
+  ([state side a b] (swap-cards-async state side nil a b))
```
Comment (NoahTheDuke) [inline src/clj/game/core/moving.clj:519]: If a function is async, it shouldn't be possible to call it incorrectly. Just remove this arity please.

File: src/clj/game/core/moving.clj [hunk 3]
```diff
@@ -518,6 +514,24 @@
...
+         install-event (= 1 (count (filter installed? [moved-a moved-b])))]
+     ;; todo - we might need behaviour for runner swap installs down the line, depending on future cards
+     ;; that's a problem for another day
+     (if (and install-event (= :corp side))
+       (trigger-event-sync
+         state side eid :corp-install
+         {:card (get-card state (if (installed? moved-a) moved-a moved-b))
+          :install-state (:install-state (card-def (if (installed? moved-a) moved-a moved-b)))})
+       (effect-completed eid))
+     async-result)))
```
Comment (NoahTheDuke) [inline src/clj/game/core/moving.clj:533]: Sadly, can't just return the async-result, you need to use `complete-with-result`: `(if (and ...) (trigger ...) (complete-with-result state side eid async-result)`

however, you also can't use `trigger-event-sync`, lol. It's been converted to a checkpoint, so you need to say (this is a mouthful, my apologies):

```clojure
(if (and install-event (= :corp side))
  (do (queue-event state :corp-install {:card ... :install-state ...})
      (wait-for (checkpoint state nil (make-eid state eid))
                (complete-with-result state side eid async-result)))
  (complete-with-result state side eid async-result))
```

File: src/clj/game/core/moving.clj [hunk 4]
```diff
@@ -510,6 +514,24 @@
         (when (in-hand? moved-b) (add-to-currently-drawing state b-side moved-b)))
       [(get-card state moved-a) (get-card state moved-b)])))
 
+(defn swap-cards-async
```
Comment (NoahTheDuke) [inline src/clj/game/core/moving.clj:517]: I don't think this is being used anywhere.

Comment (NBKelly) [reply src/clj/game/core/moving.clj:517]: Currently being used in tatu-bola to make a teia trigger when it's swapped out (I also added a test for it). 
If players find more cases where we need to use it, I can add those in later.

**Pull request merged**